libs/coin-evm/src/logic.ts

```diff
@@ -346,3 +348,37 @@ export const getMessageProperties = async (
 
   return null;
 };
+
+/**
+ * Some addresses returned by the explorers are not 40 characters hex addresses
+ * For example the explorers may return "0x0" as an address (for example for
+ * some events or contract interactions, like a contract creation transaction)
+ *
+ * This is not a valid EIP55 address and thus will fail when trying to encode it
+ * with a "Bad address" error.
+ * cf:
+ * https://github.com/cryptocoinjs/eip55/blob/v2.1.1/index.js#L5-L6
+ * https://github.com/cryptocoinjs/eip55/blob/v2.1.1/index.js#L63-L65
+ *
+ * Since we can't control what the explorer returns, and we don't want the app to crash
+ * in these cases, we simply ignore the address and return an empty string.
+ *
+ * For now this has only been observed on the from or to fields of an operation
+ * so we only use this function for these fields.
+ */
+export const safeEncodeEIP55 = (addr: string): string => {
+  if (!addr || addr === "0x" || addr === "0x0") {
+    return "";
+  }
+
+  try {
+    return eip55.encode(addr);
+  } catch (e) {
+    log("EVM Family - logic.ts", "Failed to eip55 encode address", {
+      address: addr,
+      error: e,
+    });
+
+    return addr;
```

Comment:
You changed the return value when the `eip55.encode` function failed.
It originally returned an empty string and is now replaced with the original (non-encoded) address.

>  we don't want the app to crash in these cases, we simply ignore the address and return an empty string.

Not sure we want to have badly encoded addresses flowing through the app (that's why we encode them on the edges) to avoid unexpected errors later down the road if / when these addresses are used in the codebase.

Also, could be worthwhile to add unit tests to this helper function to prevent (or at least more easily raise) these kind of changes in the future

Reply:
@lambertkevin do you valid ?

Reply:
To me there is no reason the return "" either. As a matter of fact, in those specific places "to" and "from" should *never* reach the catch. Those addresses are coming from an explorer, so the addresses can't be wrong or malformed, they're all valid Ethereum addresses. So if we had to pick a behaviour that would be to actually indeed throw if that happened as something definitely extremly wrong somewhere.

Returning nothing when you had an input is an extremly weird behaviour to me, and I'm already a bit stressed by the implication of not setting anything but an empty string in a `recipient` value of an `Operation` due to a contract deployment, which could lead to errors later on in the UI by simply inferring that "recipient *has* to be an hexadecimal string of 20 bytes" 😕 

I would personally prefer still displaying the info we got as input rather that completely removing the data if ever we had this weird error case here.

Reply:
> As a matter of fact, in those specific places "to" and "from" should never reach the catch. Those addresses are coming from an explorer, so the addresses can't be wrong or malformed, they're all valid Ethereum addresses. So if we had to pick a behaviour that would be to actually indeed throw if that happened as something definitely extremly wrong somewhere.

I agree with that, it's never supposed to happen in the _nominal_ case, but since it's data we can't control (coming from the outside of the app), it needs to be _sanityzed_ in some way to avoid issues in the app, like the original issue related to non EIP55 address returned by the backend.

But if the data is only used for display purposes in the app (possible but not sure), then we can let the original value through as is. But in this case, what's the point of encoding it to EIP 55 format in the first place then? 🤔

Reply:
> it needs to be sanityzed
I don't think it's needed, our role is not to change whatever is providing by a backend that we can't verify or update with a real value. And even in the places where we are actively using the EIP-55 checksum (e.g. when you're inputting an address yourself), we only use is as a "warning" and let the user decide if they wanna continue.

> what's the point of encoding it to EIP 55 format in the first place then? 🤔
Specifically for visual help. Here when we're encoding those strings, it can help detect an address poisining attack for example where someone will try to find a vanity address looking like another one. The end result is that by doing this checksum, you can visually detect 2 address being different even with only the first & last few caracters. 
In general, you should always have an EIP-55 encoded address displayed in the live. And like I said, here the only case where this encoding doesn't work, is when you're simply giving something else than a 20 bytes hexadecimal string, and we're now covering those cases. The rest of the try/catch is technically useless.

Reply:
In the case where the address is `0x0` or `0x` we should still return an empty string or the address as is then ?

Reply:
`""`, `0x`, it's a no brainer, yes should return an empty string.

For `0x0` is a bit different as it's not representing an empty buffer here, it's representing "0". And it might reprensent the `0x0000000000000000000000000000000000000000` burn address as in fine it's the same value. 
`0x0` == `0x0000000000000000000000000000000000000000` so that's the only case I'm not really sure. 
I think I would not add the `0x0` condition as an early return and let the try/catch do its work, unless @chabroA can confirm that this case is indeed happening on the explorers for a specific transaction like a smart contract deployment 🤷‍♂️

Reply:
Yes, explorer returns `0x0` in some cases, cf. comment (that's one reason that led me to create this function in the first place).
>  For example the explorers may return "0x0" as an address (for example for some events or contract interactions, like a contract creation transaction)

In this case, we could add a specific case for `0x0` and have the function return `0x0000000000000000000000000000000000000000` (EIP-55 encoded equivalent). Because if I remember correctly, `eip55.encode("0x0")` throws (that's why I added it in the `if` clause in the first place).

But then what do we do in the catch? If we receive an invalid address that can't be EIP-55 encoded? Do we return an empty string? Do we return the raw string address argument as is (that might not actually be an address, we don't know what it is exactly since it comes from outside the app, like from an API call)?

And if we return the raw string address argument as is, then what is the point of this function in the first place? 🤔